cpu/em_tuned.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Shriya Kulkarni <shriyak@linux.vnet.ibm.com>
+# Author: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>
+import commands
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class em_tuned(Test):
+
+    def setUp(self):
+        detected_distro = distro.detect()
+        if 'rhel' not in detected_distro.name:
+            self.cancel(" Tuned-adm service is not supported "
+                        "on %s" % detected_distro.name)
+        if SoftwareManager().check_installed("tuned") is False:
+            if SoftwareManager().install("tuned") is False:
+                self.skip("tuned is not installing")
+        self.error_count = 0
+
+    def test(self):
+        '''
+        tuned-adm is a command line tool that provides a number of different
+        profiles to improve performance in a number of specific use cases
+        Script will do basic settings of profiles using tuned-adm.
+        '''
+        self.log.info("Check for tuned service")
+        self.tuned_service_check_start()
+        self.log.info("Check for list of profiles")
+        self.check_tuned_profiles()
+        self.log.info("Changing different active profiles")
+        self.profile_change()
+        self.log.info("Check for tuned off")
+        self.check_tuned_off()
+        if self.error_count > 0:
+            self.fail(" The test case failed, check for the logs")
+
+    def tuned_service_check_start(self):
+        """
+        Check for Tuned service
+        """
+        (is_loaded, is_active) = self.get_tuned_status()
+        if is_active and is_loaded:
+            self.log.info("PASS: service tuned is running")
+        if is_loaded and not is_active:
+            self.log.info("Tuned is active not running")
+            self.log.info("Starting tuned-adm servie")
+            self.tuned_start()
+            (is_loaded, is_active) = self.get_tuned_status()
+            if not is_active:
+                self.error("Service tuned unable to start")
+            if not is_loaded:
+                self.error("Tuned is not available")
+
+    def check_tuned_profiles(self):
+        """
+        Check for Tuned profiles
+        """
+        # When ever a new profile name added to 'tuned' profiles,
+        # add the same here
+        expected_profiles = ['balanced', 'default', 'desktop-powersave',
```

Comment:
i think it can be as yaml input , it facilitate user to run any specific profile as it help to enhance the test